src/EventStore.Core/GrpcConnectionListener.cs

```diff
@@ -0,0 +1,72 @@
+using System.Collections.Generic;
+using System.Diagnostics.Metrics;
+using System.Diagnostics.Tracing;
+
+namespace EventStore.Core;
+
+public class GrpcConnectionListener : EventListener {
+	private readonly Dictionary<string, bool> _connections = new();
+	private readonly UpDownCounter<long> _grpcConnectionsMetric;
+
+	public GrpcConnectionListener(UpDownCounter<long> grpcConnectionsMetric) {
+		_grpcConnectionsMetric = grpcConnectionsMetric;
+	}
+	
+	protected override void OnEventSourceCreated(EventSource eventSource) {
+		if (eventSource.Name is not "Microsoft-AspNetCore-Server-Kestrel")
+			return;
+
+		EnableEvents(eventSource, EventLevel.Verbose);
+	}
+
+	protected override void OnEventWritten(EventWrittenEventArgs eventData) {
```

Comment:
depending on how the `Microsoft-AspNetCore-Server-Kestrel` EventSource works, i think this could be called from multiple threads simultaneously